File: test-project/greeter.startscripts/src/main/java/startscripts/DemoHelper.java
package startscripts;

import examples.greeter.api.Greeter;
import java.util.Arrays;
import java.util.ServiceLoader;
import java.util.stream.Collectors;

/**
 * Test docs
 */
public class DemoHelper {
    /**
     * Test docs
     * @param appName test docs
     * @param args test docs
     */
    public static void greet(String appName, String[] args) {
        Greeter greeter = ServiceLoader.load(Greeter.class).findFirst().orElseThrow(() -> new RuntimeException("No Greeter found!"));
        String addition = System.getProperty("greeting.addition", "");
        if(!addition.isBlank()) addition = " " + addition;
        System.out.println(appName + ": " + greeter.hello() + addition + Arrays.stream(args).collect(Collectors.joining(" and ", ", ", "!")));
    }
}

File: test-project-kotlin-pre-1-7/greeter.startscripts/src/main/kotlin/startscripts/DemoHelper.kt
package startscripts

import examples.greeter.api.Greeter
import java.util.ServiceLoader

object DemoHelper {
    fun greet(appName: String, args: Array<String>) {
        val greeter = ServiceLoader.load(Greeter::class.java)
                .findFirst()
                .orElseThrow{RuntimeException("No Greeter found!")}
        val addition = System.getProperty("greeting.addition", "").let {
            if(it.isBlank()) "" else " $it"
        }
        println("$appName: ${greeter.hello()}$addition" + args.joinToString(" and ", ", ", "!"))
    }
}
